ccl/kvccl/kvfollowerreadsccl: TestFollowerReadsWithStaleDescriptor failed

ccl/kvccl/kvfollowerreadsccl.TestFollowerReadsWithStaleDescriptor [failed](https://mesolite.cluster.engflow.com/invocations/default/82fc7ccb-23a7-4cb7-a4b1-f7e2fa39d4f4?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2t2Y2NsL2t2Zm9sbG93ZXJyZWFkc2NjbDprdmZvbGxvd2VycmVhZHNjY2xfdGVzdA==) on master @ [a0626f343f4552f375b86c45dca3091753cac13e](https://github.com/cockroachdb/cockroach/commits/a0626f343f4552f375b86c45dca3091753cac13e):


```
        	            	     0.578ms      0.000ms                [txn coordinator send: {count: 1, duration 5ms}]
        	            	     0.578ms      0.000ms                [dist sender send: {count: 1, duration 5ms}]
        	            	     0.578ms      0.000ms                [/cockroach.roachpb.KVBatch/Batch: {count: 1, duration 475µs}]
        	            	     0.607ms      0.029ms                event:sql/row/kv_batch_fetcher.go:613 [n4,client=127.0.0.1:38982,hostssl,user=root] Scan /Table/106/1/2/0
        	            	     0.616ms      0.008ms                event:sql/row/kv_batch_fetcher.go:286 [n4,client=127.0.0.1:38982,hostssl,user=root] kv fetcher: sending a batch with 1 requests
        	            	     0.618ms      0.002ms                    === operation:txn coordinator send gid:32008 _verbose:1 node:4 client:127.0.0.1:38982 hostssl: user:root txnID:9cfb1abb-36f0-4a56-acb8-42c9e4427250
        	            	     0.618ms      0.000ms                    [dist sender send: {count: 1, duration 5ms}]
        	            	     0.618ms      0.000ms                    [/cockroach.roachpb.KVBatch/Batch: {count: 1, duration 475µs}]
        	            	     0.630ms      0.013ms                        === operation:dist sender send gid:32008 _verbose:1 node:4 client:127.0.0.1:38982 hostssl: user:root txn:9cfb1abb
        	            	     0.630ms      0.000ms                        [/cockroach.roachpb.KVBatch/Batch: {count: 1, duration 475µs}]
        	            	     0.650ms      0.020ms                        event:kv/kvclient/kvcoord/range_iter.go:183 [n4,client=127.0.0.1:38982,hostssl,user=root,txn=9cfb1abb] querying next range at /Table/106/1/2/0
        	            	     0.680ms      0.030ms                        event:kv/kvclient/kvcoord/range_iter.go:220 [n4,client=127.0.0.1:38982,hostssl,user=root,txn=9cfb1abb] key: /Table/106/1/2/0, desc: r82:/{Table/106-Max} [(n1,s1):1, (n3,s3):3NON_VOTER, next=4, gen=7, sticky=9223372036.854775807,2147483647]
        	            	     0.707ms      0.026ms                        event:kv/kvclient/kvcoord/dist_sender.go:2659 [n4,client=127.0.0.1:38982,hostssl,user=root,txn=9cfb1abb] routing to nearest replica; leaseholder not required order=[(n3,s3):3NON_VOTER(health=false match=1 latency=1ms),(n1,s1):1(health=false match=1 latency=100ms)]
        	            	     0.724ms      0.018ms                        event:kv/kvclient/kvcoord/dist_sender.go:2718 [n4,client=127.0.0.1:38982,hostssl,user=root,txn=9cfb1abb] r82: sending batch 1 Get to (n3,s3):3NON_VOTER
        	            	     0.740ms      0.016ms                        event:rpc/nodedialer/nodedialer.go:230 [n4,client=127.0.0.1:38982,hostssl,user=root,txn=9cfb1abb] sending request to 127.0.0.1:42821
        	            	     4.491ms      3.752ms                        event:kv/kvclient/kvcoord/transport.go:210 [n4,client=127.0.0.1:38982,hostssl,user=root,txn=9cfb1abb] sending batch request
        	            	     4.565ms      0.073ms                            === operation:/cockroach.roachpb.KVBatch/Batch gid:32174 _verbose:1 node:3 span.kind:server request:Get [/Table/106/1/2/0], [txn: 9cfb1abb], [max_span_request_keys: 0], [target_bytes: 63230]
        	            	     4.580ms      0.016ms                            event:server/node.go:1689 [n3] node received request: 1 Get
        	            	     4.610ms      0.030ms                            event:kv/kvserver/store_send.go:158 [n3,s3] executing Get [/Table/106/1/2/0], [txn: 9cfb1abb], [max_span_request_keys: 0], [target_bytes: 63230]
        	            	     4.623ms      0.012ms                            event:kv/kvserver/replica_send.go:185 [n3,s3,r82/3:/{Table/106-Max}] read-only path
        	            	     4.637ms      0.015ms                            event:kv/kvserver/concurrency/concurrency_manager.go:291 [n3,s3,r82/3:/{Table/106-Max}] sequencing request
        	            	     4.644ms      0.006ms                            event:kv/kvserver/concurrency/concurrency_manager.go:372 [n3,s3,r82/3:/{Table/106-Max}] acquiring latches
        	            	     4.653ms      0.009ms                            event:kv/kvserver/concurrency/concurrency_manager.go:416 [n3,s3,r82/3:/{Table/106-Max}] scanning lock table for conflicting locks
        	            	     4.697ms      0.044ms                            event:kv/kvserver/replica_follower_read.go:142 [n3,s3,r82/3:/{Table/106-Max}] serving via follower read; query timestamp below closed timestamp by 1.186257619s
        	            	     4.713ms      0.015ms                            event:
        	            	     4.783ms      0.071ms                            event:kv/kvserver/replica_read.go:318 [n3,s3,r82/3:/{Table/106-Max}] can drop latches early for batch (Get [/Table/106/1/2/0], [txn: 9cfb1abb], [max_span_request_keys: 0], [target_bytes: 63230]); scanning lock table first to detect conflicts
        	            	     4.840ms      0.057ms                            event:storage/engine.go:2013 [n3,s3,r82/3:/{Table/106-Max}] lock table scan stats: seeked 1 times (1 internal); stepped 0 times (0 internal); blocks: 130B cached; points: 0 (0B keys, 0B values)
        	            	     4.849ms      0.009ms                            event:kv/kvserver/replica_read.go:121 [n3,s3,r82/3:/{Table/106-Max}] lock table scan complete without conflicts; dropping latches early
        	            	     4.865ms      0.016ms                            event:kv/kvserver/replica_read.go:431 [n3,s3,r82/3:/{Table/106-Max}] executing read-only batch
        	            	     4.988ms      0.122ms                            event:kv/kvserver/replica_evaluate.go:567 [n3,s3,r82/3:/{Table/106-Max}] evaluated Get command header:<key:"\362\211\212\210" > expect_exclusion_since:<> , txn="sql txn" meta={id=9cfb1abb key=/Min iso=Serializable pri=0.05569635 epo=0 ts=1771346831.693429000,0 min=1771346831.693429000,0 seq=0} lock=false stat=PENDING rts=1771346831.693429000,0 gul=1771346831.693429000,0 obs={n3@1771346835.898029870,0 n4@1771346835.893380466,0} : resp=header:<> , err=<nil>
        	            	     5.007ms      0.019ms                            event:n3 scan stats: stepped 0 times (0 internal); seeked 1 times (1 internal); block-bytes: (total 0 B, cached 0 B, duration 0s); points: (count 0, key-bytes 0 B, value-bytes 0 B, tombstoned: 0) ranges: (count 0), (contained-points 0, skipped-points 0) evaluated requests: 1 gets, 0 scans, 0 reverse scans
        	            	     5.018ms      0.011ms                            event:kv/kvserver/replica_read.go:243 [n3,s3,r82/3:/{Table/106-Max}] read completed
        	            	     5.039ms      0.021ms                            event:server/node.go:1799 [n3] node sending response
        	            	     4.701ms     -0.338ms                            structured:{"@type":"type.googleapis.com/cockroach.roachpb.UsedFollowerRead"}
        	            	     4.991ms      0.291ms                            structured:{"@type":"type.googleapis.com/cockroach.roachpb.ScanStats","numInterfaceSeeks":"1","numInternalSeeks":"1","numGets":"1","nodeId":3,"region":"test"}
        	            	     5.252ms      0.761ms                        event:kv/kvclient/kvcoord/transport.go:212 [n4,client=127.0.0.1:38982,hostssl,user=root,txn=9cfb1abb] received batch response
        	            	     5.299ms      4.681ms                    event:kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:617 [n4,client=127.0.0.1:38982,hostssl,user=root,txn=9cfb1abb] recording span to refresh: /Table/106/1/2/0
        	            	     5.349ms      4.775ms            event:ComponentStats{ID: {c8206475-79e3-4368-ae35-e5c25e9211b2 PROCESSOR 0 4 }, used follower read, KV nodes: n3, KV regions: test, KV time: 5ms, KV rows decoded: 0, KV bytes read: 0 B, KV gRPC calls: 1, KV pairs read: 0, MVCC step count (ext/int): 0/0, MVCC seek count (ext/int): 1/1, max memory allocated: 20 KiB, sql cpu time: 14µs, batches output: 0, rows output: 0}
        	            	     5.358ms      0.009ms            event:ComponentStats{ID: {00000000-0000-0000-0000-000000000000 UNSET 0 ? }}
        	            	     5.322ms     -0.036ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"c8206475-79e3-4368-ae35-e5c25e9211b2","type":"PROCESSOR","sqlInstanceId":4},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{"valuePlusOne":"1"},"kvPairsRead":{"valuePlusOne":"1"},"tuplesRead":{"valuePlusOne":"1"},"batchRequestsIssued":{"valuePlusOne":"2"},"kvTime":{"valuePlusOne":"0.004738926s"},"contentionTime":{"valuePlusOne":"0.000000001s"},"lockWaitTime":{"valuePlusOne":"0.000000001s"},"latchWaitTime":{"valuePlusOne":"0.000000001s"},"numInterfaceSteps":{"valuePlusOne":"1"},"numInternalSteps":{"valuePlusOne":"1"},"numInterfaceSeeks":{"valuePlusOne":"2"},"numInternalSeeks":{"valuePlusOne":"2"},"blockBytes":{"valuePlusOne":"1"},"blockBytesInCache":{"valuePlusOne":"1"},"keyBytes":{"valuePlusOne":"1"},"valueBytes":{"valuePlusOne":"1"},"pointCount":{"valuePlusOne":"1"},"pointsCoveredByRangeTombstones":{"valuePlusOne":"1"},"rangeKeyCount":{"valuePlusOne":"1"},"rangeKeyContainedPoints":{"valuePlusOne":"1"},"rangeKeySkippedPoints":{"valuePlusOne":"1"},"kvCpuTime":{},"numGets":{"valuePlusOne":"2"},"numScans":{"valuePlusOne":"1"},"numReverseScans":{"valuePlusOne":"1"},"nodeIds":[3],"regions":["test"],"usedFollowerRead":true},"exec":{"execTime":{},"maxAllocatedMem":{"valuePlusOne":"20481"},"maxAllocatedDisk":{},"consumedRu":{"valuePlusOne":"1"},"cpuTime":{"valuePlusOne":"0.000013631s"}},"output":{"numBatches":{"valuePlusOne":"1"},"numTuples":{"valuePlusOne":"1"}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
        	            	     5.353ms      0.031ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"00000000-0000-0000-0000-000000000000"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
        	            	     5.383ms      4.809ms        event:ComponentStats{ID: {c8206475-79e3-4368-ae35-e5c25e9211b2 FLOW 0 4 test}}
        	            	     5.379ms     -0.004ms        structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"c8206475-79e3-4368-ae35-e5c25e9211b2","type":"FLOW","sqlInstanceId":4,"region":"test"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{"valuePlusOne":"20481"},"maxDiskUsage":{"valuePlusOne":"1"},"consumedRu":{}}}
        	            	     5.444ms      5.044ms    event:sql/conn_executor_exec.go:2160 [n4,client=127.0.0.1:38982,hostssl,user=root] execution ends
        	            	     5.457ms      0.013ms    event:sql/conn_executor_exec.go:2160 [n4,client=127.0.0.1:38982,hostssl,user=root] rows affected: 0
        	            	     5.499ms      0.043ms        === operation:commit sql txn gid:32008 _verbose:1 node:4 client:127.0.0.1:38982 hostssl: user:root
        	            	     5.525ms      0.068ms    event:sql/conn_executor_exec.go:4304 [n4,client=127.0.0.1:38982,hostssl,user=root] AutoCommit. err: <nil>
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestFollowerReadsWithStaleDescriptor2165520320
--- FAIL: TestFollowerReadsWithStaleDescriptor (5.83s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFollowerReadsWithStaleDescriptor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60467

ccl/kvccl/kvfollowerreadsccl.TestFollowerReadsWithStaleDescriptor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21117392?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21117392?buildTab=artifacts#/) on master @ [9eb28243edfe56c067525195999d1637622a0da0](https://github.com/cockroachdb/cockroach/commits/9eb28243edfe56c067525195999d1637622a0da0):


```
        	            	     0.614ms      0.000ms                [dist sender send: {count: 1, duration 9ms}]
        	            	     0.614ms      0.000ms                [/cockroach.roachpb.KVBatch/Batch: {count: 2, duration 4ms}]
        	            	     0.652ms      0.039ms                event:sql/row/kv_batch_fetcher.go:613 [n4,client=127.0.0.1:42434,hostssl,user=root] Scan /Table/106/1/2/0
        	            	     0.661ms      0.009ms                event:sql/row/kv_batch_fetcher.go:286 [n4,client=127.0.0.1:42434,hostssl,user=root] kv fetcher: sending a batch with 1 requests
        	            	     0.665ms      0.004ms                    === operation:txn coordinator send gid:24944 _verbose:1 node:4 client:127.0.0.1:42434 hostssl: user:root txnID:63ee5fb1-7108-474b-a34a-06da5f0f8cc7
        	            	     0.665ms      0.000ms                    [dist sender send: {count: 1, duration 9ms}]
        	            	     0.665ms      0.000ms                    [/cockroach.roachpb.KVBatch/Batch: {count: 2, duration 4ms}]
        	            	     0.677ms      0.011ms                        === operation:dist sender send gid:24944 _verbose:1 node:4 client:127.0.0.1:42434 hostssl: user:root txn:63ee5fb1
        	            	     0.677ms      0.000ms                        [/cockroach.roachpb.KVBatch/Batch: {count: 2, duration 4ms}]
        	            	     0.698ms      0.021ms                        event:kv/kvclient/kvcoord/range_iter.go:183 [n4,client=127.0.0.1:42434,hostssl,user=root,txn=63ee5fb1] querying next range at /Table/106/1/2/0
        	            	     0.739ms      0.041ms                        event:kv/kvclient/kvcoord/range_iter.go:220 [n4,client=127.0.0.1:42434,hostssl,user=root,txn=63ee5fb1] key: /Table/106/1/2/0, desc: r82:/{Table/106-Max} [(n1,s1):1, (n3,s3):3NON_VOTER, next=4, gen=7, sticky=9223372036.854775807,2147483647]
        	            	     0.770ms      0.032ms                        event:kv/kvclient/kvcoord/dist_sender.go:2659 [n4,client=127.0.0.1:42434,hostssl,user=root,txn=63ee5fb1] routing to nearest replica; leaseholder not required order=[(n3,s3):3NON_VOTER(health=false match=1 latency=1ms),(n1,s1):1(health=false match=1 latency=100ms)]
        	            	     0.793ms      0.023ms                        event:kv/kvclient/kvcoord/dist_sender.go:2718 [n4,client=127.0.0.1:42434,hostssl,user=root,txn=63ee5fb1] r82: sending batch 1 Get to (n3,s3):3NON_VOTER
        	            	     0.811ms      0.018ms                        event:rpc/nodedialer/nodedialer.go:230 [n4,client=127.0.0.1:42434,hostssl,user=root,txn=63ee5fb1] sending request to 127.0.0.1:34647
        	            	     7.384ms      6.573ms                        event:kv/kvclient/kvcoord/transport.go:210 [n4,client=127.0.0.1:42434,hostssl,user=root,txn=63ee5fb1] sending batch request
        	            	     7.392ms      0.008ms                            === operation:/cockroach.roachpb.KVBatch/Batch gid:24944 _verbose:1 node:4 client:127.0.0.1:42434 hostssl: user:root txn:63ee5fb1 span.kind:client
        	            	     7.610ms      0.218ms                            === operation:/cockroach.roachpb.KVBatch/Batch gid:25701 _verbose:1 node:3 span.kind:server request:Get [/Table/106/1/2/0], [txn: 63ee5fb1], [max_span_request_keys: 0], [target_bytes: 54386]
        	            	     7.649ms      0.039ms                            event:server/node.go:1689 [n3] node received request: 1 Get
        	            	     7.754ms      0.105ms                            event:kv/kvserver/store_send.go:158 [n3,s3] executing Get [/Table/106/1/2/0], [txn: 63ee5fb1], [max_span_request_keys: 0], [target_bytes: 54386]
        	            	     8.928ms      1.174ms                            event:kv/kvserver/replica_send.go:185 [n3,s3,r82/3:/{Table/106-Max}] read-only path
        	            	     8.954ms      0.026ms                            event:kv/kvserver/concurrency/concurrency_manager.go:291 [n3,s3,r82/3:/{Table/106-Max}] sequencing request
        	            	     8.963ms      0.009ms                            event:kv/kvserver/concurrency/concurrency_manager.go:372 [n3,s3,r82/3:/{Table/106-Max}] acquiring latches
        	            	     8.975ms      0.012ms                            event:kv/kvserver/concurrency/concurrency_manager.go:416 [n3,s3,r82/3:/{Table/106-Max}] scanning lock table for conflicting locks
        	            	     9.011ms      0.036ms                            event:kv/kvserver/replica_follower_read.go:142 [n3,s3,r82/3:/{Table/106-Max}] serving via follower read; query timestamp below closed timestamp by 1.185271661s
        	            	     9.030ms      0.020ms                            event:
        	            	     9.150ms      0.119ms                            event:kv/kvserver/replica_read.go:318 [n3,s3,r82/3:/{Table/106-Max}] can drop latches early for batch (Get [/Table/106/1/2/0], [txn: 63ee5fb1], [max_span_request_keys: 0], [target_bytes: 54386]); scanning lock table first to detect conflicts
        	            	     9.190ms      0.040ms                            event:storage/engine.go:2013 [n3,s3,r82/3:/{Table/106-Max}] lock table scan stats: seeked 1 times (1 internal); stepped 0 times (0 internal)
        	            	     9.199ms      0.009ms                            event:kv/kvserver/replica_read.go:121 [n3,s3,r82/3:/{Table/106-Max}] lock table scan complete without conflicts; dropping latches early
        	            	     9.216ms      0.017ms                            event:kv/kvserver/replica_read.go:431 [n3,s3,r82/3:/{Table/106-Max}] executing read-only batch
        	            	     9.364ms      0.148ms                            event:kv/kvserver/replica_evaluate.go:567 [n3,s3,r82/3:/{Table/106-Max}] evaluated Get command header:<key:"\362\211\212\210" > expect_exclusion_since:<> , txn="sql txn" meta={id=63ee5fb1 key=/Min iso=Serializable pri=0.00431044 epo=0 ts=1771370634.369303000,0 min=1771370634.369303000,0 seq=0} lock=false stat=PENDING rts=1771370634.369303000,0 gul=1771370634.369303000,0 obs={n3@1771370638.576984917,0 n4@1771370638.569236117,0} : resp=header:<> , err=<nil>
        	            	     9.403ms      0.040ms                            event:n3 scan stats: stepped 0 times (0 internal); seeked 1 times (1 internal); block-bytes: (total 0 B, cached 0 B, duration 0s); points: (count 0, key-bytes 0 B, value-bytes 0 B, tombstoned: 0) ranges: (count 0), (contained-points 0, skipped-points 0) evaluated requests: 1 gets, 0 scans, 0 reverse scans
        	            	     9.422ms      0.018ms                            event:kv/kvserver/replica_read.go:243 [n3,s3,r82/3:/{Table/106-Max}] read completed
        	            	     9.454ms      0.033ms                            event:server/node.go:1799 [n3] node sending response
        	            	     9.017ms     -0.438ms                            structured:{"@type":"type.googleapis.com/cockroach.roachpb.UsedFollowerRead"}
        	            	     9.373ms      0.356ms                            structured:{"@type":"type.googleapis.com/cockroach.roachpb.ScanStats","numInterfaceSeeks":"1","numInternalSeeks":"1","numGets":"1","nodeId":3,"region":"test"}
        	            	     9.843ms      2.459ms                        event:kv/kvclient/kvcoord/transport.go:212 [n4,client=127.0.0.1:42434,hostssl,user=root,txn=63ee5fb1] received batch response
        	            	     9.913ms      9.247ms                    event:kv/kvclient/kvcoord/txn_interceptor_span_refresher.go:617 [n4,client=127.0.0.1:42434,hostssl,user=root,txn=63ee5fb1] recording span to refresh: /Table/106/1/2/0
        	            	     9.994ms      9.385ms            event:ComponentStats{ID: {71e4c096-79ce-4c7f-94ca-7e76923a92ec PROCESSOR 0 4 }, used follower read, KV nodes: n3, KV regions: test, KV time: 9ms, KV rows decoded: 0, KV bytes read: 0 B, KV gRPC calls: 1, KV pairs read: 0, MVCC step count (ext/int): 0/0, MVCC seek count (ext/int): 1/1, max memory allocated: 20 KiB, sql cpu time: 18µs, batches output: 0, rows output: 0}
        	            	    10.000ms      0.006ms            event:ComponentStats{ID: {00000000-0000-0000-0000-000000000000 UNSET 0 ? }}
        	            	     9.956ms     -0.044ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"71e4c096-79ce-4c7f-94ca-7e76923a92ec","type":"PROCESSOR","sqlInstanceId":4},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{"valuePlusOne":"1"},"kvPairsRead":{"valuePlusOne":"1"},"tuplesRead":{"valuePlusOne":"1"},"batchRequestsIssued":{"valuePlusOne":"2"},"kvTime":{"valuePlusOne":"0.009327564s"},"contentionTime":{"valuePlusOne":"0.000000001s"},"lockWaitTime":{"valuePlusOne":"0.000000001s"},"latchWaitTime":{"valuePlusOne":"0.000000001s"},"numInterfaceSteps":{"valuePlusOne":"1"},"numInternalSteps":{"valuePlusOne":"1"},"numInterfaceSeeks":{"valuePlusOne":"2"},"numInternalSeeks":{"valuePlusOne":"2"},"blockBytes":{"valuePlusOne":"1"},"blockBytesInCache":{"valuePlusOne":"1"},"keyBytes":{"valuePlusOne":"1"},"valueBytes":{"valuePlusOne":"1"},"pointCount":{"valuePlusOne":"1"},"pointsCoveredByRangeTombstones":{"valuePlusOne":"1"},"rangeKeyCount":{"valuePlusOne":"1"},"rangeKeyContainedPoints":{"valuePlusOne":"1"},"rangeKeySkippedPoints":{"valuePlusOne":"1"},"kvCpuTime":{},"numGets":{"valuePlusOne":"2"},"numScans":{"valuePlusOne":"1"},"numReverseScans":{"valuePlusOne":"1"},"nodeIds":[3],"regions":["test"],"usedFollowerRead":true},"exec":{"execTime":{},"maxAllocatedMem":{"valuePlusOne":"20481"},"maxAllocatedDisk":{},"consumedRu":{"valuePlusOne":"1"},"cpuTime":{"valuePlusOne":"0.000018004s"}},"output":{"numBatches":{"valuePlusOne":"1"},"numTuples":{"valuePlusOne":"1"}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
        	            	     9.996ms      0.040ms            structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"00000000-0000-0000-0000-000000000000"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{},"maxDiskUsage":{},"consumedRu":{}}}
        	            	    10.033ms      9.426ms        event:ComponentStats{ID: {71e4c096-79ce-4c7f-94ca-7e76923a92ec FLOW 0 4 test}}
        	            	    10.029ms     -0.005ms        structured:{"@type":"type.googleapis.com/cockroach.sql.distsqlrun.ComponentStats","component":{"flowId":"71e4c096-79ce-4c7f-94ca-7e76923a92ec","type":"FLOW","sqlInstanceId":4,"region":"test"},"netRx":{"latency":{},"waitTime":{},"deserializationTime":{},"tuplesReceived":{},"bytesReceived":{},"messagesReceived":{}},"netTx":{"tuplesSent":{},"bytesSent":{},"messagesSent":{}},"kv":{"bytesRead":{},"kvPairsRead":{},"tuplesRead":{},"batchRequestsIssued":{},"kvTime":{},"contentionTime":{},"lockWaitTime":{},"latchWaitTime":{},"numInterfaceSteps":{},"numInternalSteps":{},"numInterfaceSeeks":{},"numInternalSeeks":{},"blockBytes":{},"blockBytesInCache":{},"keyBytes":{},"valueBytes":{},"pointCount":{},"pointsCoveredByRangeTombstones":{},"rangeKeyCount":{},"rangeKeyContainedPoints":{},"rangeKeySkippedPoints":{},"kvCpuTime":{},"numGets":{},"numScans":{},"numReverseScans":{}},"exec":{"execTime":{},"maxAllocatedMem":{},"maxAllocatedDisk":{},"consumedRu":{},"cpuTime":{}},"output":{"numBatches":{},"numTuples":{}},"flowStats":{"maxMemUsage":{"valuePlusOne":"20481"},"maxDiskUsage":{"valuePlusOne":"1"},"consumedRu":{}}}
        	            	    10.074ms      9.616ms    event:sql/conn_executor_exec.go:1154 [n4,client=127.0.0.1:42434,hostssl,user=root] execution ends
        	            	    10.084ms      0.010ms    event:sql/conn_executor_exec.go:1154 [n4,client=127.0.0.1:42434,hostssl,user=root] rows affected: 0
        	            	    10.155ms      0.071ms        === operation:commit sql txn gid:24944 _verbose:1 node:4 client:127.0.0.1:42434 hostssl: user:root
        	            	    10.194ms      0.110ms    event:sql/conn_executor_exec.go:4304 [n4,client=127.0.0.1:42434,hostssl,user=root] AutoCommit. err: <nil>
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/18f40673c89ae073fd9b4688a4790099/logTestFollowerReadsWithStaleDescriptor1132140166
--- FAIL: TestFollowerReadsWithStaleDescriptor (5.75s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFollowerReadsWithStaleDescriptor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvtestutils/test_utils.go
+++ b/pkg/kv/kvtestutils/test_utils.go
@@ -20,7 +20,8 @@ import (
 func OnlyFollowerReads(rec tracingpb.Recording) bool {
 	foundFollowerRead := false
 	for _, sp := range rec {
-		if sp.Operation != "/cockroach.roachpb.Internal/Batch" {
+		if sp.Operation != "/cockroach.roachpb.Internal/Batch" &&
+			sp.Operation != "/cockroach.roachpb.KVBatch/Batch" {
 			continue
 		}
 		anonTagGroup := sp.FindTagGroup(tracingpb.AnonymousTagGroupName)

